include/cuco/detail/static_map/static_map.inl

```diff
@@ -318,11 +320,47 @@ void static_map<Key, T, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Stora
   auto const num = cuco::detail::distance(first, last);
   if (num == 0) { return; }
 
-  auto const grid_size = cuco::detail::grid_size(num, cg_size);
-
-  static_map_ns::detail::insert_or_apply<cg_size, cuco::detail::default_block_size()>
-    <<<grid_size, cuco::detail::default_block_size(), 0, stream.get()>>>(
-      first, num, op, ref(op::insert_or_apply));
+  int32_t constexpr shmem_block_size = 1024;
+  int32_t const default_grid_size    = cuco::detail::grid_size(num, cg_size);
+
+  int32_t constexpr cardinality_threshold   = shmem_block_size;
+  int32_t constexpr shared_map_num_elements = cardinality_threshold + shmem_block_size;
+  float constexpr load_factor               = 0.7;
+  int32_t constexpr shared_map_size         = (1.0 / load_factor) * shared_map_num_elements;
+
+  using extent_type            = cuco::extent<int32_t, shared_map_size>;
+  using shared_map_type        = cuco::static_map<Key,
+                                                  T,
+                                                  extent_type,
+                                                  cuda::thread_scope_block,
+                                                  KeyEqual,
+                                                  ProbingScheme,
```

Comment:
Is the shared map using the same probing scheme and CG size as the global memory map? IIRC shared memory hash maps perform best with a CG size of 1.

Reply:
The current implemenation supports shmem only with with CG size = 1. When CG != 1 it always fallbacks to global memory map insert_or_apply.

Reply:
I see. That's fine for now. We can always revisit this topic in the future